spot-contracts/contracts/FeePolicy.sol

```diff
@@ -0,0 +1,259 @@
+ *              rolloverFeePerc = sigmoid(normalizedSubscriptionRatio) / rotationsPerYear
+ *
+ */
+contract FeePolicy is IFeePolicy, OwnableUpgradeable {
+    // Libraries
+    using MathUpgradeable for uint256;
+    using SafeCastUpgradeable for uint256;
+
+    /// @dev The returned fee percentages are fixed point numbers with {DECIMALS} places.
+    ///      The decimals should line up with value expected by consumer (perp).
+    ///      NOTE: 10**DECIMALS => 100% or 1.0
+    uint8 public constant DECIMALS = 8;
+    uint256 public constant ONE = (1 * 10**DECIMALS); // 1.0 or 100%
+    uint256 public constant SIGMOID_BOUND = ONE / 10; // 0.10 or 10%
+
+    /// @dev Number of seconds in one year. (365.25 * 24 * 3600)
+    int256 public constant ONE_YEAR_SEC = 31557600;
+
+    /// @notice Reference to the perpetual token.
+    IPerpetualTranche public perp;
+
+    /// @notice Reference to the rollover vault.
+    IVault public vault;
+
+    /// @notice The target deviation i.e) the normalization factor.
+    uint256 public targetSubscriptionRatio;
+
+    /// @notice The perp token's mint fee percentage ceiling.
+    uint256 public maxPerpMintFeePerc;
+
+    /// @notice The perp token's burn fee percentage ceiling.
+    uint256 public maxPerpBurnFeePerc;
+
+    /// @notice The vault note's mint fee percentage ceiling.
+    uint256 public maxVaultMintFeePerc;
+
+    /// @notice The vault note's burn fee percentage ceiling.
+    uint256 public maxVaultBurnFeePerc;
+
+    /// @notice The fixed amount vault fee withheld during each deployment.
+    /// @dev Denominated in the underlying collateral asset and
+    ///      Paid by the vault note holders to the system owner.
+    uint256 public vaultDeploymentFee;
+
+    struct SigmoidParams {
+        /// @notice Lower asymptote
+        int256 lower;
+        /// @notice Upper asymptote
+        int256 upper;
+        /// @notice sigmoid slope
+        int256 growth;
+    }
+
+    /// @notice Parameters which control the asymptotes and the slope of the yearly perp token's rollover fee.
+    SigmoidParams public perpRolloverFeeAPR;
+
+    /// @dev The current subscription state the vault system relative to the perp supply.
+    struct SubscriptionState {
+        /// @notice The current tvl of perp.
+        uint256 currentPerpTVL;
+        /// @notice The current tvl of the rollover vault.
+        uint256 currentVaultTVL;
+        /// @notice The target tvl of the rollover vault.
+        uint256 targetVaultTVL;
+        /// @notice Computed normalized subscription ratio.
+        uint256 normalizedSubscriptionRatio;
+    }
+
+    /// @notice Contract initializer.
+    /// @param perp_ Reference to perp.
+    /// @param vault_ Reference to the rollover vault.
+    function init(IPerpetualTranche perp_, IVault vault_) public initializer {
+        __Ownable_init();
+
+        perp = perp_;
+        vault = vault_;
+
+        maxPerpMintFeePerc = ONE / 30; // 0.033
```

Comment:
Should we just init everything to 0? I assume whatever these are will be explicitly updated after deployment anyway.

I like having some non-zero values for the curve params, though

Reply:
Sure 👍